docs/authentication.md

```diff
@@ -102,13 +102,13 @@ Each compute should present a token valid for the timeline's tenant.
 Pageserver also has HTTP API: some parts are per-tenant,
 some parts are server-wide, these are different scopes.
 
-The `auth_type` configuration variable in Pageserver's config may have
-either of three values:
+The `auth_validation_public_key_path` config option, if present, enables JWT validation.
```

Comment:
Maybe not in the config files, but I definitely want to replace `(auth_type, auth_validation_public_key_path)` with a single value, be it `enum` or `Option` or whatnot.

Reply:
> In this security business it seems to me that being explicit is better

+1

Also in theory we can have different auth types with the same key

> with a single value

Can be auth_config or something like that. enum or plain struct. In doc comment we can explain the semantics. 

AFAIK this parameter is not explicitly set in config, so we should be able to avoid format migration. (worth double checking)